tests/unit/Store-ObservableArray.js

```diff
@@ -0,0 +1,510 @@
+			}));
+			store10.source = new ObservableArray(
+				{ id: 1, name: "itam 1", number: 8 },
+				{ id: 2, name: "item 2", number: 4 },
+				{ id: 3, name: "itam 3", number: 8 },
+				{ id: 4, name: "item 4", number: 5 },
+				{ id: 5, name: "itam 5", number: 8 },
+				{ id: 6, name: "item 6", number: 5 },
+				{ id: 7, name: "itam 7", number: 8 },
+				{ id: 8, name: "item 8", number: 5 },
+				{ id: 9, name: "itam 9", number: 8 });
+			var store11 = new C();
+			store11.query = new Filter().lte("number", 5).gte("number", 5);
+			store11.deliver();
+			store11.on("query-success", d.callback(function () {
+				assert.strictEqual(store11.renderItems.length, 3, "multi query");
+			}));
+			store11.source = new ObservableArray(
+				{ id: 1, name: "itam 1", number: 8 },
+				{ id: 2, name: "item 2", number: 4 },
+				{ id: 3, name: "itam 3", number: 8 },
+				{ id: 4, name: "item 4", number: 5 },
+				{ id: 5, name: "itam 5", number: 8 },
+				{ id: 6, name: "item 6", number: 5 },
+				{ id: 7, name: "itam 7", number: 8 },
+				{ id: 8, name: "item 8", number: 5 },
+				{ id: 9, name: "itam 9", number: 8 });
+			return d;
+		},
+
+		StoreFuncRange: function () {
+			var d = this.async(1500);
+			var store = new C();
+			store.fetch = function (collection) {
+				return collection.fetchRange({start: 0, end: 1});
+			};
+			store.on("query-success", d.callback(function () {
+				assert(store.renderItems instanceof Array);
+				assert.strictEqual(store.renderItems.length, 1);
+			}));
+			store.source = new ObservableArray({ id: "foo", name: "Foo" },
+				{ id: "bar", name: "Bar" });
+			return d;
+		},
+
+		StoreFuncSort: function () {
+			var d = this.async(1500);
+			var store = new C();
+			store.processQueryResult = function (store) {
+				return store.sort(function (a, b) {
+					if (a.index > b.index) {
+						return 1;
+					}
+					if (a.index < b.index) {
+						return -1;
+					}
+					return 0;
+				});
+			};
+			var myData = [
+				{ id: "foo", name: "Foo", index: 1 },
+				{ id: "bar", name: "Bar", index: 0 }
+			];
+			store.on("query-success", d.callback(function () {
+				assert(store.renderItems instanceof Array);
+				assert.strictEqual(store.renderItems.length, 2);
+				assert.deepEqual(store.renderItems[0], myData[1]);
+				assert.deepEqual(store.renderItems[1], myData[0]);
+				//var item = store.source[0];
+				//item.index = 2;
+				//store.source.push(item);
+				//store.deliver();
+				//// this works because put is synchronous
+				//assert.deepEqual(store.renderItems[0], { id: "bar", name: "Bar", index: 1 });
+				//assert.deepEqual(store.renderItems[1], { id: "foo", name: "Foo", index: 2 });
+				//item = store.source[1];
+				//item.index = 0;
+				//store.source.push(item);
+				//store.deliver();
+				//assert.deepEqual(store.renderItems[0], { id: "foo", name: "Foo", index: 0 });
+				//assert.deepEqual(store.renderItems[1], { id: "bar", name: "Bar", index: 1 });
```

Comment:
Because of the difference explain here : https://github.com/harbalk/delite/commit/1d9adf043fbf8547577d77bfb5a4dd5303318280#commitcomment-12636199.
Maybe I should remove the comments ?